examples/Generic/README.md

```diff
@@ -3,20 +3,145 @@
 This directory contains a minimal example that shows how to use `AstarteGenericDevice`.
 
 The device is instantiated in `ExampleDevice.java`, the code is commented to highlight all the
-things needeed to create your Astarte Device.
+things needed to create your Astarte Device.
+
+
+## How To
+
+Download
+--------
+
+### Gradle
+
+Add to "dependencies" (as for any module you need):
+
+``` groovy
+dependencies {
+    implementation 'org.astarte-platform:devicesdk-generic:$version'
+}
+```
+
+Usage
+-----
+### 1. Device registration and credentials secret emission
+*If your device was already register, you can skip this section.*
+
+You must register the device beforehand to obtain its `credentials-secret`.
+You have three options to achieve this:
+1. Through [`astartectl`](https://github.com/astarte-platform/astartectl) command.
+2. Through Astarte Dashboard of your instance, which is located at `https://dashboard.<your-astarte-domain>`.
+3. Through `AstartePairingService` class contained in sdk (see following paragraph for details).
+
+#### 1.1 Programmatically Generate Astarte Device id
+You can generate an astarte device id either by random or by combining namespace and unique data.
+
+##### 1.1.1 Random
+``` java
+String deviceId = AstarteDeviceIdUtils.generateId();
+```
+
+##### 1.1.2 Namespace and unique data
+``` java
+UUID uuidNameSpace = UUID.fromString("f79ad91f-c638-4889-ae74-9d001a3b4cf8");
+String deviceId = AstarteDeviceIdUtils.generateId(uuidNameSpace, "001122334455");
```

Comment:
let's use an example MAC address, so our users can understand how to use it from a practical usecase.